viking1.js

```diff
@@ -0,0 +1,101 @@
+// Soldier
+class Soldier {
+    constructor(health, strength) {
+      this.health = health;
+      this.strength = strength;
+    }
+  
+    attack() {
+      return this.strength;
+    }
+  
+    receiveDamage(damage) {
+      this.health -= damage;
+    }
+  }
+
+// Viking
+class Viking extends Soldier {
+  constructor(name , health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage;

```

Comment:
Aqui você pode usar o super e chamar `super.receiveDamage(damage)` ao invés de escrever novamente `this.health -= damage`.